message/messaging_service.cc

```diff
@@ -1032,6 +1033,16 @@ future<> messaging_service::send_gossip_digest_ack2(msg_addr id, gossip_digest_a
     return send_message_oneway(this, messaging_verb::GOSSIP_DIGEST_ACK2, std::move(id), std::move(msg));
 }
 
+void messaging_service::register_gossip_get_endpoint_states(std::function<future<gms::gossip_get_endpoint_states_response> (const rpc::client_info& cinfo, gms::gossip_get_endpoint_states_request request)>&& func) {
+    register_handler(this, messaging_verb::GOSSIP_GET_ENDPOINT_STATES, std::move(func));
+}
+future<> messaging_service::unregister_gossip_get_endpoint_states() {
+    return unregister_handler(messaging_verb::GOSSIP_GET_ENDPOINT_STATES);
+}
+future<gms::gossip_get_endpoint_states_response> messaging_service::send_gossip_get_endpoint_states(msg_addr id, gms::gossip_get_endpoint_states_request request) {
+    return send_message_timeout<future<gms::gossip_get_endpoint_states_response>>(this, messaging_verb::GOSSIP_GET_ENDPOINT_STATES, std::move(id), 5000ms, std::move(request));
```

Comment:
It's better to expose the timeout. messaging_service shouldn't decide on what is the right timeout for gossip.

Reply:
OK.